### PR TITLE
added hpmm

### PR DESCRIPTION
**Pull request by jchenWill**
added human performance monitoring module for NASA task 3 and uploaded a de-identified dataset from task 3.

### CODE REVIEW

**Comment by jmcfarland-swri** [comment]
Thank you for the contribution. I will review and provide some feedback.

In general, it is best to create a branch for your work instead of using the "master" branch. This way, you can pull changes from the upstream repository into your repository without worrying about conflicting with the work that you are doing on your branch. It also allows you to work on multiple things at once, each under a separate branch. If you want to move your work onto a branch, I can help you do this, but a new pull request will need to be submitted.

In order to make revisions to the code to address comments from the review, you can simply push new commits to your branch, and they will automatically be reflected in the pull request.

**Comment by jmcfarland-swri** [comment]
I'm going to go ahead and review, but I want to reiterate that you will be better off in the long run if you submit your pull request using a branch other than the master branch. The reason I am pointing this out now is that changing branches will require a completely new pull request.

If you're new to git, we have a lot of resources on our [Contributing](https://github.com/ymlasu/para-atm/wiki/Contributing) page, in particular the online Git Book linked at the top.

**Comment by jmcfarland-swri** [comment]
It looks like you have added various data files that are not required by your module and are not used by any of the testing.  For example, `catboost_training.json`, `events.out.tfevents`, `learn_error.tsv`, etc.  Please remove these.  The only files that should be added to para-atm are the source code for your module and any minimal sample data files that are used as part of tests or examples.

File: paraatm/HPMM/hpmm.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+This Python script is used for HPMM (Human Performance Monitoring Module)
+The script outputs predicted loss of separation as an indicator of air traffic controller's performance
+The script can also check compliance for each command
+@Last Modified by: Jiawei Chen
+@Last Modified date: 2020-05-26
+"""
+from typing import Any, Union
+
+from jpype import *
+from array import *
```
Comment (jmcfarland-swri) [inline paraatm/HPMM/hpmm.py:16]: Please remove use of `import *`, which makes it difficult to know which packages the functions that you use in your code come from, or which of your imports are actually used in the code.

File: paraatm/HPMM/hpmm.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+import matplotlib.pyplot as plt
+from pandas import Series, DataFrame
+from pandas.io.parsers import TextFileReader
+from scipy.stats import norm
+from sklearn.preprocessing import StandardScaler
+from sklearn.model_selection import train_test_split
+import catboost as cb
+from catboost import CatBoostRegressor
+from catboost import Pool
```
Comment (jmcfarland-swri) [inline paraatm/HPMM/hpmm.py:35]: Please remove all imports that are not actually used in this file.  Based on a quick check, it appears that most of these (e.g., `sts`, `os`, `time`, `sys`, `datetime`, etc.) are not used.

File: paraatm/HPMM/hpmm.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+
+
+class HPMM(object):
+    def __init__(self, cfg):
+        self.data = pd.read_csv(cfg['datafile']);
```
Comment (jmcfarland-swri) [inline paraatm/HPMM/hpmm.py:41]: Why have you chosen to use a dictionary with one key as the input here?  Why not just pass in the name of the data file as a string, or even better pass in array data structures with the data?  Having the function accept one or more array data structures will be more consistent with how other data analysis methods are designed in Python, and it avoids the need to make assumptions about the format of this data file that is being provided by the user.

Also, since this is a "public" function that users of your module would interact with, please add a docstring that includes at a minimum describes the function parameters.  See for example [iff.py](../blob/master/io/iff.py).

File: paraatm/HPMM/hpmm.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        data_domain.fillna(data_domain.mean(), inplace=True)
+
+        # transform categorical data
+        data_domain = data_domain.join(pd.get_dummies(data_domain.sa_correct))
+        data_domain = data_domain.join(pd.get_dummies(data_domain['query']))
+        data_domain = data_domain.drop(columns=['query', 'sa_correct'])
+        return data_domain
+
+    def modelpredict(model, input):
+        return model.predict(input)
```
Comment (jmcfarland-swri) [inline paraatm/HPMM/hpmm.py:61]: Please follow the convention of naming the first method argument `self`.  Also, I haven't tried to test the code yet, but I would be surprised if it runs: the `HPMM` class does not have a `predict` method, so it's unclear what should happen here when `modelpredict` is called on an instance of the `HPMM` class.

File: paraatm/HPMM/hpmm.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+    def train(self, model, train_pool, test_pool, plot):
+        model.fit(train_pool, eval_set=test_pool, plot=plot)
+
+
+if __name__ == '__main__':
+    # my code
+    cfg = {
+        'datafile': '/home/jchen311/para-atm/paraatm/sample_data/human_data.csv'
+    }
```
Comment (jmcfarland-swri) [inline paraatm/HPMM/hpmm.py:71]: Please remove this reference to your local file system.  This will not work on anyone's system but yours.  See [test_para_atm.py](../blob/master/paraatm/tests/test_para_atm.py) for examples of the best practice for locating files in the `sample_data` directory in a universal way.

**Comment by jmcfarland-swri** [comment]
Please carefully review the sections in the [Contributing Guide](https://github.com/ymlasu/para-atm/wiki/Contributing) on "Organization of the code" and "Style".  In particular, directory (sub-package) names should be all lowercase.

**Comment by jmcfarland-swri** [comment]
It appears that the purpose of your `HPMM` class is only data pre-processing.  Is that correct?  If that is the case, it would make more sense not to use a class and instead just define a regular function:  it accepts some kind of input and returns some kind of output.  There is no need for it to be in a class.

File: paraatm/HPMM/hpmm.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        data_domain = data_domain.join(pd.get_dummies(data_domain.sa_correct))
+        data_domain = data_domain.join(pd.get_dummies(data_domain['query']))
+        data_domain = data_domain.drop(columns=['query', 'sa_correct'])
+        return data_domain
+
+    def modelpredict(model, input):
+        return model.predict(input)
+
+    def train(self, model, train_pool, test_pool, plot):
+        model.fit(train_pool, eval_set=test_pool, plot=plot)
```
Comment (jmcfarland-swri) [inline paraatm/HPMM/hpmm.py:64]: Similar comment as with `modelpredict`.  Your `HPMM` class does not actually have a `fit` method, so calling `train` on an instance of `HPMM` would not work.  I'm wondering if maybe this is just part of some incomplete work?

**Comment by jmcfarland-swri** [comment]
We have published the PARA-ATM Research Collection on GitHub:
https://github.com/ymlasu/para-atm-collection

This contribution would fit better in the Research Collection, as opposed to the para-atm Python package.  This will be the best way to demonstrate what it does without having to worry about all of the details of documentation, API, test cases, and integration with other code.

Please take a look at the Guide to Contributing for the Research Collection:
https://github.com/ymlasu/para-atm-collection/wiki/Contributing
It is much simpler than for the para-atm Python package, as you do not need to worry about forks, branches, or pull requests.  You should have access to submit contributions directly.

It would be good to revisit the previous comments from this pull request and consider the suggestions for cleanup of the code, but it is not necessary to follow the specific coding style conventions when submitting to the Research Collection.  However, some of the comments should be addressed, such as eliminating the reference to local files (`/home/jchen311/para-atm/paraatm/sample_data/human_data.csv`) so that others are able to run the example code directly.

Once it is ready, my suggestion would be to convert the code under `
if __name__ == '__main__':` in  `hpmm.py` into a Jupyter notebook.  That way, you can add explanation in the notebook of the example and what the various steps in the code are doing.  It would also be good to add a plot to the end of the example, e.g. a cross-validation plot of actual versus predicted values for the test set.  You can save the generated figures as part of the notebook as well – then, users can view the example on GitHub and see exactly what it does without having to install or run anything.  You can see some of the existing examples in the Research Collection:
https://github.com/ymlasu/para-atm-collection/blob/master/miscellaneous/para-atm-demo/paraatm-demo.ipynb
https://github.com/ymlasu/para-atm-collection/blob/master/miscellaneous/weather/echotop.ipynb
https://github.com/jmcfarland-swri/ULI-example/blob/master/info-fusion/svm-text-classifier/svm_text_classifier_example.ipynb

Let us know if you have questions or if we can help get you started.
